### PR TITLE
Feature/plcn 328 create authorization behaviour

### PR DESCRIPTION
**Pull request by AntonEsbjergIT-Minds**
Created authorization behaviour. The point is that we have a authorize tag that can be set on any class and if that calss is used the current user has to have the correct authorization as set in the tag. If the tag is not on a class nothing happens, if the tag is on a class it is checked if the current user has an userId if they don't they are unauthorized, if they have an userId their roles are compared to the roles need to view the class if they do not have a requiste role access is forbidden.

### CODE REVIEW

File: src/Pelican.Application/Authentication/ForbiddenAccessException.cs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+﻿namespace Pelican.Application.Authentication;
+public class ForbiddenAccessException : Exception
```
Comment (Nicolai-Bak) [inline src/Pelican.Application/Authentication/ForbiddenAccessException.cs:2]: Is this necessary or could we use `UnauthorizedAccessException`

File: .config/dotnet-tools.json [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+      ]
+    },
+    "snitch": {
+      "version": "1.12.0",
+      "commands": [
+        "snitch"
+      ]
+    }
+  }
 }
```
Comment (Nicolai-Bak) [inline .config/dotnet-tools.json:18]: is this intended?

File: src/Pelican.App/Pelican.App.csproj [hunk 3]
```diff
@@ -9,10 +9,10 @@
...
 	<ItemGroup>
-		<PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="6.0.10">
-			<PrivateAssets>all</PrivateAssets>
-			<IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
-		</PackageReference>
+	  <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="6.0.10">
+	    <PrivateAssets>all</PrivateAssets>
+	    <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+	  </PackageReference>
 	</ItemGroup>
```
Comment (Nicolai-Bak) [inline src/Pelican.App/Pelican.App.csproj:16]: is this intended?

File: src/Pelican.Application/DependencyInjection.cs [hunk 4]
```diff
@@ -37,6 +37,8 @@ public static IServiceCollection AddApplication(this IServiceCollection services
 
 		services.AddScoped(typeof(IPipelineBehavior<,>), typeof(ValidationBehaviour<,>));
 
+		services.AddScoped(typeof(IPipelineBehavior<,>), typeof(AuthorizationBehaviour<,>));
+
```
Comment (Nicolai-Bak) [inline src/Pelican.Application/DependencyInjection.cs:41]: In my opinion it makes more sense to add AuthorizationBehaviour before ValidationBehaviour. 

GetCustomAttributesService should be added as a service before AuthorizationBehaviour.

Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Application/DependencyInjection.cs:41]: Switched the two behavious and added GetCustomAttributesService

File: src/Pelican.Application/Authentication/ForbiddenAccessException.cs (hunk 1 above)
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.Application/Authentication/ForbiddenAccessException.cs:2]: Probably not I will remove it.

File: .config/dotnet-tools.json (hunk 2 above)
Comment (AntonEsbjergIT-Minds) [reply .config/dotnet-tools.json:18]: Nope, fixed it.

File: src/Pelican.App/Pelican.App.csproj (hunk 3 above)
Comment (AntonEsbjergIT-Minds) [reply src/Pelican.App/Pelican.App.csproj:16]: Nope fixed it.

**Review by Nicolai-Bak — APPROVED** [review]

**Review by s-bogelund — APPROVED** [review]

**Review by itm-mne — APPROVED** [review]

**Pull request merged**